tensorboard/pip_package/build_pip_package.sh

```diff
@@ -94,140 +83,39 @@ cp -LR "${RUNFILES}/org_tensorflow_serving_api/tensorflow_serving" tensorboard/_
-  # logdir does not contain any checkpoints and thus an empty runs.
-  grep '\[\]' projector_runs.json
-  curl -fs "http://localhost:$(cat port)/data/plugin/projector/projector_binary.html" >projector_binary.html
-  grep '<vz-projector-dashboard' projector_binary.html
-  kill $!
-
-  # Test TensorBoard APIs
-  export TF_CPP_MIN_LOG_LEVEL=1  # Suppress spammy TF startup logging.
-  python -c "
-import tensorboard as tb
-assert tb.__version__ == tb.version.VERSION
-from tensorboard.plugins.projector import visualize_embeddings
-tb.notebook.start  # don't invoke; just check existence
-from tensorboard.plugins.hparams import summary_v2 as hp
-hp.hparams_pb({'optimizer': 'adam', 'learning_rate': 0.02})
-"
-  if [ -n "${smoke_tf}" ]; then
-    # Only test summary scalar, beholder, and mesh summary
-    python -c "
-import tensorboard as tb
-tb.summary.scalar_pb('test', 42)
-from tensorboard.plugins.beholder import Beholder, BeholderHook
-from tensorboard.plugins.mesh import summary
-"
-  fi
-
-  if [ -n "${smoke_tf}" ]; then
-    # Exhaustively test various sequences of importing tf.summary.
-    test_tf_summary() {
-      # First argument is subpath to test, e.g. '' or '.compat.v2'.
-      import_attr="import tensorflow as tf; a = tf${1}.summary; a.write; a.scalar"
-      import_as="import tensorflow${1}.summary as b; b.write; b.scalar"
-      import_from="from tensorflow${1} import summary as c; c.write; c.scalar"
-      printf '%s\n' "${import_attr}" "${import_as}" "${import_from}" | python -
-      printf '%s\n' "${import_attr}" "${import_from}" "${import_as}" | python -
-      printf '%s\n' "${import_as}" "${import_attr}" "${import_from}" | python -
-      printf '%s\n' "${import_as}" "${import_from}" "${import_attr}" | python -
-      printf '%s\n' "${import_from}" "${import_attr}" "${import_as}" | python -
-      printf '%s\n' "${import_from}" "${import_as}" "${import_attr}" | python -
-    }
-    test_tf_summary '.compat.v2'
-    is_tf_2() {
-      python -c "import tensorflow as tf; assert tf.__version__[:2] == '2.'" \
-        >/dev/null 2>&1
-    }
-    if is_tf_2 ; then
-      test_tf_summary ''
-    fi
-  fi
-
-  deactivate
-  cd ..
-  rm -rf "${smoke_venv}"
-}
-
-case "${smoke}" in
-  "all")
-    smoke 2 "${tf_version}"
-    smoke 3 "${tf_version}"
-    ;;
-  "1")
-    # Empty string indicates to use the default "python".
-    smoke "" "${tf_version}"
-    ;;
-  "0")
-    printf "\nSkipping smoke test\n\n"
+cd "${original_wd}"  # Bazel gives "${output}" as a relative path >_>
+case "${output}" in
+  *.tar.gz)
+    mkdir -p "$(dirname "${output}")"
+    "${RUNFILES}/org_tensorflow_tensorboard/tensorboard/pip_package/deterministic_tar_gz" \
+        "${output}" "${workdir}"/dist/*.whl
     ;;
   *)
-    echo >&2 'fatal: unknown smoke value:' "${smoke}"
-    exit 1
+    if ! [ -d "${output}" ]; then
+      printf >&2 'fatal: no such output directory: %s\n' "${output}"
```

Comment:
Seems more straightforward to just `mkdir -p "${output}"` here if it doesn't exist.  I don't see any particular reason why it should require the directory to already exist, and this makes it more consistent with the `.tar.gz` behavior?

Reply:
Requiring the output directory to exist is more consistent with other applications
that write files to specified directories—

```
$ tar czf - a.c | tar xzf - -C ./nonexistent/
tar: nonexistent: Cannot open: No such file or directory
tar: Error is not recoverable: exiting now
$ cp a.c ./nonexistent/
cp: cannot create regular file './nonexistent/': Not a directory
$ mv a.c ./nonexistent/
mv: cannot move 'a.c' to './nonexistent/': Not a directory
$ ln -s a.c ./nonexistent/
ln: target './nonexistent/' is not a directory: No such file or directory
$ gcc a.c -o ./nonexistent/
/usr/bin/ld: cannot open output file ./nonexistent/: Is a directory
collect2: error: ld returned 1 exit status
$ gcc a.c -o ./nonexistent/a.c
/usr/bin/ld: cannot open output file ./nonexistent/a.c: No such file or directory
collect2: error: ld returned 1 exit status
$ printf 'shell redirections\n' >./nonexistent/out
bash: ./nonexistent/out: No such file or directory
```

—so I’d prefer to keep it as is unless you feel strongly. (The explicit
error is just to have a nicer message.)